### ISSUE TITLE
Server not found, only when logged in to myplex

### ISSUE THREAD

**Issue by i814u2** [iBaa/PlexConnect#390]
Current with master, as of this commit: 1d4e285.
aTV3, PlexConnect on a Mac mini, PlexMediaServer on a linux server

As soon as I upgraded, I exited Trailers on the aTV, then went back into Trailers. It said it couldn't find a media server (I didn't note the exact error and forgot it now).

I checked logs and found that it wasn't authenticating. It worked the other day, but probably had remained logged in since I last changed my password. So I logged out of MyPlex and when it refreshed the screen, it went from 0 servers to 1. 
Regardless, I entered my login information. It refreshed but showed 0 again. So I logged out in order to watch some shows.

Afterward, I checked the log file and you can see the authentication issue but then it simply failed to return anything from the "poke" of plex.tv.
I checked my XML (from the same machine running PlexConnect), via this url: https://plex.tv/api/resources?includeHttps=1
And it looks fine. It shows both the internal and external server info. Also: publicAddressMatches="1",httpsRequired="0",productVersion="0.9.12.16.1509-8e31408"

This is the log file: http://pasted.co/6f7252a1

If I can provide any other information, please let me know. This seemed different from issue #389, but feel free to merge if you think otherwise.

**Comment by iBaa** [comment]
I am confused - is it now working for you or not? Comment in #389 suggest it is...
If not, please provide a log of the issue with loglevel="High". It will contain more information, mandatory to understand what's going on in the details. Thanks!

**Comment by i814u2** [comment]
Sorry for being unclear. I logged out, and logged back in but it did not fix anything for me. I logged back out in order to watch some shows.

I'll test again when I get home and report back. I just wanted to clarify that initially I tried to log out/in and it did NOT help.

**Comment by i814u2** [comment]
Tonight I enabled High logging in the config and reloaded the plist on the mac mini serving PlexConnect.

After doing so, I attempted to login to MyPlex. Server count went to 0 (it was 1, and working, before logging in to MyPlex). I manually attempted to discover servers but it still said 0. I logged out and it went back to 1 and we were able to watch a show without issue.

This is the log: http://pasted.co/2392e9a9 . I removed anything that I thought may be sensitive. I redacted items I knew to search (my usernames, server names, IP addresses, etc), and things that I found near those known items (various tokens, udid, etc). If I went a bit overboard, and you need more information, please let me know and I can review the sections more specifically. 

Basically, I still appear to have an issue when I'm logged in to MyPlex, but my local server works if I'm not logged in (and only if I'm not logged in to MyPlex).

**Comment by Rycochet** [comment]
Without looking through could it be that you might have Secure Connections set to Required rather than Preferred?

**Comment by i814u2** [comment]
I just checked the Plex server and it is set at "Preferred" (under the Network settings). That's the only spot I could find with that setting. If there is more than one spot to check, please let me know.

**Comment by iBaa** [comment]
DNS rebind protection, as discussed here: https://github.com/iBaa/PlexConnect/issues/391 ?

**Comment by i814u2** [comment]
That sounds possible. I will investigate later today.

I have noticed something odd about my ssh tunnel from work, so perhaps it's a similar issue, or at least related in some way. I'll do some additionally testing around the network routing settings and report back with any findings.

**Comment by i814u2** [comment]
I checked my router logs and it mentioned a "possible" dns-rebind attack detected. However, nothing about blocking it.

Since I was logged out of MyPlex anyway, I added the proper config change to dnsmasq (I use pfsense for my router setup). That change should have added plex.direct to a whitelist, as I understand it.

Then I logged in to MyPlex and it found the server. So I thought it was working and decided to do my normal validation by undoing the most recent step (the dnsmasq config) and testing again. In this case, it should have failed. However, it still worked. I assumed maybe something was cached, so I restarted PlexConnect but it still worked. 
I did notice that it did not load any thumbnails. So I tried various combinations of restarting PlexConnect, adding/removing the plex.direct entry from the router, upgrading PlexConnect, logging in and out of MyPlex, etc. 

So I'm currently on commit bc48d83, my router does NOT have plex.direct added to the config, and everything is working fine. Basically, I'm back to my old config and everything is working. I have no idea why this failed for me and then started working. Unless there is some cache portion I'm not aware of at this time.

If there is some PlexConnect cache somewhere, let me know and I can clear it to attempt to re-create the issue. Otherwise this can be closed. I can't re-produce the issue now for some reason. I'll check to see if there is a cache portion in pfsense that would affect this, but it seems unlikely (unless it's actually cached in the DNS, since it was allowed through once). Router details aren't foreign to me, but they aren't my strongest knowledge base. I'll keep checking around the router to see what I can find.

Again, if this seems pointless, we can close this issue. If you'd prefer to get a more definitive answer, I'm willing to help continue to track it down.

**Comment by i814u2** [comment]
CONFIRMED: My issues were due to my router protecting against DNS re-binding attacks (similar to #391).
For anyone seeing this later, if you're testing something, the DNS forwarder (dnsmasq) caches information and the service needs to be restarted in order for changes to be properly tested.
Additionally, I had to log out and back in to MyPlex in order to fully test. I cleared the cache on the PlexConnect machine first, but it didn't appear to matter. So either something in the aTV is caching, or it is PlexConnect (more likely, I would guess).

Anyway, I proved it was the rebinding protection by clearing all caches mentioned above, removing any whitelisting form my router, logging out of MyPlex, and then logging back in. It found 0 servers. Then I added the proper line to the DNS Forwarding config and clicked on the discover line of PlexConnect settings. It found my server.

For folks with pfSense, here is the site I found with info:
https://doc.pfsense.org/index.php/DNS_Rebinding_Protections

To make your setup work, navigate to the DNS Forwarder (under the Services section/tab, depending on your theme), and enter this text in the Advanced section (click the Advanced button if needed):
`rebind-domain-ok=/plex.direct/`

iBaa - How would you prefer to handle these issues? Do you happen to have any tricks in mind to avoid this situation, or is it pointless since we're dealing with DNS redirects in the first place? I'll wait for a response before closing the issue, just in case you have something you want to track here.
